src/GitWrapper.cc

```diff
@@ -0,0 +1,401 @@
+#include <git2.h>
+#include "taskolib/GitWrapper.h"
+#include "taskolib/exceptions.h"
+#include "gul14/cat.h"
+#include <iostream>
+#include <vector>
+
+
+LibGit::LibGit(std::filesystem::path file_path)
+{
+    //init libgit library
+    git_libgit2_init();
+
+    
+    //init member variables
+    repo_path_ = file_path;
+
+    // if repository does not exist
+    if (git_repository_open(repo_.getptr(), (repo_path_).c_str())) 
```

Comment:
The `getptr()` solution is dangerous because it allows `git_repository_open()` to modify the private internals of `repo_`. I would suggest to write a wrapper for `git_repository_open()`. Something like this:
```cpp
GitPointer<git_repository> open_repository(const std::string& repo_path)
{
    git_repository* ptr = nullptr;
    if (git_repository_open(&ptr, (repo_path_).c_str()) == OK)
        return ptr;
    else
       return nullptr;
}
```